### PR TITLE
Bug 1624773 - fix CatalogSource and ClusterServiceVersion link on InstallPlans page

### PR DESCRIPTION
**Pull request by yapei**
- Fixes CS link on InstallPlansDetails page
- Fixes CSV link on InstallPlansLists page

@alecmerdler  Could you help review? Thanks very much

### CODE REVIEW

**Comment by coreos-ui** [comment]
Can one of the admins verify this patch?

**Comment by coreos-ui** [comment]
Can one of the admins verify this patch?

**Comment by yapei** [comment]
retest

**Comment by xiaocwan** [comment]
retest

**Comment by yapei** [comment]
I see a failure in spec test `__tests__/components/cloud-services/install-plan.spec.tsx:72:76`, will fix

File: frontend/public/components/cloud-services/install-plan.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ export const InstallPlanRow: React.SFC<InstallPlanRowProps> = (props) => {
       <ResourceLink kind="Namespace" name={props.obj.metadata.namespace} title={props.obj.metadata.namespace} displayName={props.obj.metadata.namespace} />
     </div>
     <div className="hidden-xs col-sm-4 col-md-3 col-lg-2">
-      {props.obj.spec.clusterServiceVersionNames.map((csvName, i) => <span key={i}><ResourceIcon kind={referenceForModel(ClusterServiceVersionModel)} /> {csvName}</span>)}
+      {props.obj.spec.clusterServiceVersionNames.map((csvName, i) => <span key={i}><ResourceLink kind={referenceForModel(ClusterServiceVersionModel)} name={csvName} namespace={props.obj.metadata.namespace} /></span>)}
```
Comment (alecmerdler) [inline frontend/public/components/cloud-services/install-plan.tsx]: Use the same [logic as in `InstallPlanDetails`](https://github.com/openshift/console/blob/master/frontend/public/components/cloud-services/install-plan.tsx#L88) to make sure we don't link to a `ClusterServiceVersion` that doesn't exist.

File: frontend/public/components/cloud-services/install-plan.tsx [hunk 2]
```diff
@@ -91,7 +91,7 @@ export const InstallPlanDetails: React.SFC<InstallPlanDetailsProps> = ({obj}) =>
               </dd>) }
               <dt>Catalog Sources</dt>
               { (_.get(obj.status, 'catalogSources') || []).map((catalogName, i) => <dd key={i}>
-                <ResourceLink kind={referenceForModel(CatalogSourceModel)} name={catalogName} namespace="openshift" title={catalogName} />
+                <ResourceLink kind={referenceForModel(CatalogSourceModel)} name={catalogName} namespace={obj.spec.sourceNamespace || olmNamespace} title={catalogName} />
```
Comment (alecmerdler) [inline frontend/public/components/cloud-services/install-plan.tsx]: Nice, good catch!

File: frontend/public/components/cloud-services/install-plan.tsx (hunk 1 above)
Comment (yapei) [reply frontend/public/components/cloud-services/install-plan.tsx]: Requested changes updated

**Comment by yapei** [comment]
Seems a common failure in backend tests

**Comment by spadgett** [comment]
This looks like a legitimate failure

```
ERROR in /go/src/github.com/openshift/console/frontend/__tests__/components/operator-lifecycle-manager/install-plan.spec.tsx
(11,37): 'ResourceIcon' is declared but its value is never read.
```

**Comment by yapei** [comment]
Because of unfamiliar with contribution workflow, I didn't give a complete test locally before submitting PR, so I'm going close the PR and create a new one

**Comment by spadgett** [comment]
@yapei fyi, you can always push updates to your branch without needing to open a new PR

**Pull request closed without merging**